api/web_team_hub.go

```diff
@@ -96,7 +96,8 @@ func ShouldSendEvent(webCon *WebConn, msg *model.Message) bool {
 		}
 	} else {
 		// Don't share a user's view events with other users
-		if msg.Action == model.ACTION_CHANNEL_VIEWED {
+		// but you still need to share it with yourself
+		if webCon.UserId != msg.UserId && msg.Action == model.ACTION_CHANNEL_VIEWED {
```

Comment:
This is unnecessary since if it gets into the enclosing `else` it's not the same user